### PR TITLE
feat(image): discover SBOM in OCI referrers

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Discover SBOM in the OCI referrers via [the referrers tag](https://github.com/opencontainers/distribution-spec/blob/acfc11dad63159052f98dd9afab04adf59e6ed8f/spec.md#referrers-tag-schema) or [API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers).

### Usage

```
$ trivy image --sbom-sources oci ghcr.io/knqyf263/oci-referrers
2023-03-05T17:36:55.278+0200    INFO    Vulnerability scanning is enabled
2023-03-05T17:36:58.103+0200    INFO    Detected SBOM format: cyclonedx-json
2023-03-05T17:36:58.129+0200    INFO    Found SBOM (cyclonedx) in the OCI referrers
2023-03-05T17:36:58.133+0200    INFO    Detected OS: alpine
2023-03-05T17:36:58.133+0200    INFO    Detecting Alpine vulnerabilities...
2023-03-05T17:36:58.134+0200    INFO    Number of language-specific files: 1
2023-03-05T17:36:58.134+0200    INFO    Detecting gobinary vulnerabilities...

ghcr.io/knqyf263/oci-referrers (alpine 3.16.2)
==============================================
Total: 17 (UNKNOWN: 0, LOW: 0, MEDIUM: 5, HIGH: 9, CRITICAL: 3)
```

## TODO
- [x] Discover SBOM in OCI referrers
- [ ] Support private registries (in another PR)
- [x] Write tests
- [x] Update documents

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/3735

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: docs/docs/target/container_image.md [hunk 1]
```diff
@@ -375,10 +375,39 @@ $ skopeo copy docker-daemon:alpine:3.11 oci:/path/to/alpine
...
-## SBOM generation
+## SBOM
+Trivy supports the generation of Software Bill of Materials (SBOM) for container images and the search for SBOMs during vulnerability scanning.
+
+### Generation
 Trivy can generate SBOM for container images.
 See [here](../sbom/index.md) for the detail.
 
+### Discovery
+Trivy can search for Software Bill of Materials (SBOMs) that reference container images.
```
Comment (AnaisUrlichs) [inline docs/docs/target/container_image.md:386]: maybe specify in this sentence where the 'search' happens?

Comment (knqyf263) [reply docs/docs/target/container_image.md:386]: >maybe specify in this sentence where the 'search' happens?

This page is for container images (`docs/docs/target/container_image.md`). Do you mean something else?

> then why would we want trivy to be able to discover the SBOMs related to that image

Added https://github.com/aquasecurity/trivy/pull/3768/commits/537dcea300ed0dfbc6f1ff53f8f9bc869c8a5ab7. Thanks.

File: docs/docs/target/container_image.md [hunk 2]
```diff
@@ -375,10 +375,41 @@ $ skopeo copy docker-daemon:alpine:3.11 oci:/path/to/alpine
...
+2023-03-05T17:36:58.103+0200    INFO    Detected SBOM format: cyclonedx-json
+2023-03-05T17:36:58.129+0200    INFO    Found SBOM (cyclonedx) in the OCI referrers
+...
+
+ghcr.io/knqyf263/oci-referrers (alpine 3.16.2)
+==============================================
+Total: 17 (UNKNOWN: 0, LOW: 0, MEDIUM: 5, HIGH: 9, CRITICAL: 3)
+```
+
+The OCI Registry utilizes the Referrers API (https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers).
```
Comment (otms61) [inline docs/docs/target/container_image.md:410]: Do you intend an embedded URL?
```suggestion
The OCI Registry utilizes the [Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers).
```

File: pkg/oci/artifact.go [hunk 3]
```diff
@@ -18,7 +18,22 @@ import (
 	"github.com/aquasecurity/trivy/pkg/downloader"
 )
 
-const titleAnnotation = "org.opencontainers.image.title"
+const (
+	// Artifact types
+	CycloneDXArtifactType = "application/vnd.cyclonedx+json"
+	SPDXArtifactType      = "application/vnd.spdx+json"
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/oci/artifact.go]: Good catch! Fixed in https://github.com/aquasecurity/trivy/pull/3768/commits/41beeab00e0e2b05b3eb815fb2c59ae517275890.

**Review by otms61 — APPROVED** [review]

**Comment by Preisschild** [comment]
From the PR

>  Support private registries (in another PR)

@knqyf263 Hey, do you know if this was already implemented?

**Pull request merged**